Adds section on transforming content. (#483)
mdx-js/mdx: docs/advanced/transform-content.md

# Transform Content

> As we previously stated, you will most likely use MDX through a loader or framework, but it is also possible to interact with or manipulate raw MDX content.  Since MDX is a part of the [unified][unified] ecosystem, there are many utilities we can use to work with MDX.

Reviewer: ```suggestion
With MDX you can interact with or manipulate raw MDX content.  Since MDX is a part of the [unified][unified] ecosystem, there are many utilities you can use to work with MDX.
```

> We can utilize [to-vfile][to-vfile] to read and write MDX files and we can leverage [remark][remark] and [remark-mdx][remark-mdx] to parse and process MDX content.

Reviewer: ```suggestion
You can utilize [to-vfile][to-vfile] to read and write MDX files and you can leverage [remark][remark] and [remark-mdx][remark-mdx] to parse and process MDX content.
```

The following example demonstrates how to put all of that together.

```js
const {read, write} = require('to-vfile')
const remark = require('remark');
const mdx = require('remark-mdx');

(async () => {
    const path = './path/to/some/file.mdx';

>     const file = await read();

Reviewer: ```suggestion
    const file = await read(path);
```

    const contents = await remark()
        .use(mdx)

>         // use more plugins or write your own here

Reviewer: Should we show an example of plugin usage and then display input/output to make it clear what's going on?

        .process(file);

    await write({
        path,
        contents
    })
})();
```

> As the comment in the code above points out, you can use any of the [remark plugins][remark-plugins], or even write your own, to interpolate MDX content.  Through the use of remark-mdx, it will enhance the [AST][ast] to understand MDX, giving you access and insight to things like imports and exports.

Reviewer: ```suggestion
As the comment in the code above points out, you can use any of the [remark plugins][remark-plugins], or even write your own, to interpolate MDX content.  The remark-mdx library is a parsing extension to enhance the Markdown [AST][ast] to understand MDX (resulting in MDXAST), giving you access and insight to things like imports and exports.
```
